src/cart/crt_hg.c

```diff
@@ -832,14 +834,12 @@ crt_rpc_handler_common(hg_handle_t hg_hdl)
 			  "failed to invoke RPC handler, rc: %d, opc: %#x\n",
 			  rc, opc);
 		crt_hg_reply_error_send(rpc_priv, rc);
+		HG_Destroy(rpc_priv->crp_hg_hdl);
+		D_GOTO(decref, hg_ret = HG_SUCCESS);
 	}
 
 decref:
-	/* If rpc call back is customized, then it might be handled
-	 * asynchronously, Let's hold the RPC, and the real handler
-	 * (crt_handle_rpc())will release it
-	 */
-	if (rc != 0 || !crt_rpc_cb_customized(crt_ctx, &rpc_priv->crp_pub))
```

Comment:
ok, the previous crt_rpc_cb_customized() check was to differentiate daos usage and IOF/CPPR usage before, fine to remove it now we only have daos.

Reply:
I moved this in other place for not customized path.